src/python/WMCore/Storage/TrivialFileCatalog.py

```diff
@@ -173,20 +175,55 @@ def __str__(self):
                 result += "\n"
         return result
 
-
-def tfcProtocol(contactString):
+def tfcProtocol(contactString,useTFC):
     """
     _tfcProtocol_
 
-    Given a Trivial File Catalog contact string, extract the
-    protocol from it.
+    Given a file catalog contact string, extract the
+    protocol from it. If useTFC=True, the Trivial File Catalog contract string is used (storage.xml). Otherwise, it is the file catalog from Rucio definitions (storage.json) 
 
     """
     args = urlsplit(contactString)[3]
     value = args.replace("protocol=", '')
+    if not useTFC:
+        value = value.split('&')[0]
     return value
+def getCatalog(storageAttr):
+    """
+    Construct a catalog for storage.json from storage attributes. It is a string with the same format as old trivial catalog for storage.xml ('trivialcatalog_file:'+path+'?protocol='+storageAttr['protocol']+'&volume='+storageAttr['volume']. "path" points to location of storage.json, '/abc/xyz/storage.json')  
+    """
```

Comment:
About the cross site terminology, yes it is similar to fallback in the previous mechanism where the storage is directed to other site rather than the "current site" as we can see in this:
https://gitlab.cern.ch/SITECONF/T2_DE_DESY/-/blob/master/JobConfig/site-local-config.xml
the fallback is pointed to <phedex-node value="T2_DE_RWTH"/>
However, in the new mechanism, there can be multiple "method" in <stage-out> and one or many of them can pointed to other sites so I use "cross site". There is only one fallback. I am not sure there are requirements for close proximity of these disk storage.
We can see an example of multiple <stage-out> at above config as well

The subsite is special in term of relation with site. It has no disk storage but use disk storage of site so it belongs to site in term of disk storage. Therefore, subsite does not fit the "cross site" term.
This is an example of subsite:
https://gitlab.cern.ch/SITECONF/T1_DE_KIT

I made suggest changes